### PR TITLE
GH-1306 Adds new icon/text pins to map/Pins.js

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-1306

http://rawgit.com/jbrundage/Visualization/MapIconTextPins/demos/dermatology.html?src/map/Layered&zoom=2

@GordonSmith Please review

Signed-off-by: jay <jbrundage372@gmail.com>

### CODE REVIEW

File: test/mapFactory.js [hunk 1]
```diff
@@ -258,7 +258,7 @@
         },
         Layered: {
             simple: function (callback) {
-                createMap(false, [], null, callback);
+                createMap(true, [], null, callback);
```
Comment (GordonSmith) [inline test/mapFactory.js]: This switches the OpenMap demo to use Google Maps?

File: src/map/Pins.js [hunk 2]
```diff
@@ -29,16 +30,52 @@
     Pins.prototype.layerUpdate = function (base) {
         Layer.prototype.layerUpdate.apply(this, arguments);
 
-        this._pinsTransform
-            .style("opacity", this.opacity())
-        ;
```
Comment (GordonSmith) [inline src/map/Pins.js]: This is still needed?

File: src/map/Pins.js [hunk 3]
```diff
@@ -15,6 +15,7 @@
 
     Pins.prototype.publish("opacity", 1.0, "number", "Opacity", null, { tags: ["Advanced"] });
 
+    Pins.prototype.publish("pinShape", "square", "set", "Shape of pin", ["square","circle"]);
```
Comment (GordonSmith) [inline src/map/Pins.js]: Can we keep the current "small" pin and call it shape "pin"?

File: src/map/Pins.js [hunk 4]
```diff
@@ -29,16 +30,52 @@
...
-        this._pinsTransform
-            .style("opacity", this.opacity())
-        ;
-
         this.pinsPaths = this._pinsTransform.selectAll(".data").data(this.visible() ? this.data() : [], function (d) { return d[0]; });
         var context = this;
         this.pinsPaths.enter().append("path")
             .attr("class", "data")
             .attr("d", function (d) {
-                return "M 0,0 C -2,-20 -10,-22 -10,-30 A 10,10 0 1,1 10,-30 C 10,-22 2,-20 0,0 z M -2,-30";
```
Comment (GordonSmith) [inline src/map/Pins.js]: Lets keep this as shape === "pin" (as it will be useful having a small pin)

**Comment by jbrundage** [comment]
@GordonSmith I've made your suggested changes, but the Travis CI build keeps failing because of src/google/MaterialGantt

**Comment by GordonSmith** [comment]
@jbrundage I fixed that this morning, so you can rebase onto latest candidate-1.14.0 and it will be fixed  (Google renamed the chart from GanttChart to Gantt)

File: src/map/Pins.js [hunk 5]
```diff
@@ -15,6 +15,7 @@
 
     Pins.prototype.publish("opacity", 1.0, "number", "Opacity", null, { tags: ["Advanced"] });
 
+    Pins.prototype.publish("pinShape", "square", "set", "Shape of pin", ["circle","pin","square"]);
```
Comment (GordonSmith) [inline src/map/Pins.js]: Minor:  Should default to pin for backward compatability

File: src/map/Pins.js [hunk 6]
```diff
@@ -28,49 +29,127 @@
...
                 var pos = base.project(d[0], d[1]);
                 if (!pos) {
                     pos = [0, 0];
                 }
-                return "translate(" + pos[0] + ", " + pos[1] + ")scale(" + 0.5 + ")";
+                d3.select(".pin-icon-"+i).attr("transform","translate("+pos[0]+", "+pos[1]+")scale("+0.5+")");
+                d3.select(".pin-text-"+i).attr("transform","translate("+pos[0]+", "+pos[1]+")scale("+0.5+")");
+                return "translate(" + pos[0] + ", " + pos[1] + ")scale(" + 1 /* * base._zoom.scale() */ + ")";
             })
+            .attr("stroke-width", 1 + "px")
```
Comment (GordonSmith) [inline src/map/Pins.js]: FYI - Anything inside "layerZoomed" is very performance sensitive - in general we get away with simply applying a transform on the outer layer (for geohash and choropleths), but pins need to keep their size no matter the zoom level, hence each one gets the transformation.

**Comment by GordonSmith** [comment]
Rebase and fix default to Pin and I can pull, thx.

**Pull request closed without merging**